crates/block-producer/src/store/mod.rs

```diff
@@ -53,15 +50,6 @@ pub trait Store: ApplyBlock {
         produced_nullifiers: impl Iterator<Item = &Nullifier> + Send,
         notes: impl Iterator<Item = &NoteId> + Send,
     ) -> Result<BlockInputs, BlockInputsError>;
-
-    /// Returns note authentication information for the set of specified notes.
-    ///
-    /// If authentication info for a note does not exist in the store, the note is omitted
-    /// from the returned set of notes.
-    async fn get_note_authentication_info(
-        &self,
-        notes: impl Iterator<Item = &NoteId> + Send,
-    ) -> Result<NoteAuthenticationInfo, NotePathsError>;
```

Comment:
Is this no longer needed? I thought we'd still need to use it during batch building - no?

Reply:
Iiuc we have several opportunities to fetch note inclusion proofs:

1. Transaction proving (client side)
2. Transaction inputs (currently we just fetch whether or not they're present I think)
3. Batch proving
4. Block proving

Of these, only (4) absolutely has to happen to complete the final proof?

And of these, only (3) has no other database IO at all aside from this so I thought maybe its somewhat pointless. As in, the rest all have to reach out to the database for other data as well, so may as well amortize the network/disk IO.

I'll add it back though; no solid argument against it :)

Reply:
Re-adding this I did run into [this](https://github.com/0xPolygonMiden/miden-node/blob/573f37b0fd7cd572889cdf2ce20623495e421f3f/crates/block-producer/src/errors.rs#L103) error description, which makes it sound like I need an accurate list to request from the store?

Previously this wasn't an issue because we effectively had the entire batch set to query against, whereas now we only have this one batch's data. i.e. we don't know which notes are inflight still, and which are committed in the store?

If we want to do this check here then we need to either allow the store lookup to be partial, or to perform some additional bookkeeping somehow.

Reply:
What I wanted to do was change `GetTransactionInputs` to include the note inclusion proofs. And then batch building doesn't reach out the store at all, and finally only the block builder gets the outstanding since it can remove ephemeral notes.

Reply:
The scenario that we want to handle here is roughly like this:

1. Transaction `A` which consumes an unauthenticated note `x` arrives to the block producer. At this point, `x` is in-flight.
2. During transaction validation, we go to the store but since `x` is in-flight, there is nothing new the store can tell us about `x`.
3. Transaction which creates note `x` gets included in the block and persisted to the store.
4. Then, transaction `A` gets selected for batching.

At this point, we want to "upgrade" `x` to an authenticated note. For this, we need to get its authentication path and there are 2 ways to do it:

- We can get the path from the store. This is how it worked before.
- We can attach authentication path to unauthenticated notes as a part of step 3 above (i.e., when the new block gets built).

The latter approach could be more efficient as it saves us a trip to the store, but I think it would also require quite a bit more bookkeeping and has potential for more bugs. So, for now, I think the former approach (i.e., going to the store for authentication info) is probably the simplest solution.

And if we need to update how the store endpoint works for this, I think that's fine.

Reply:
Okay I'll leave this PR as is and re-introduce this in another PR with store changes as needed.